0-common/schema-ddl/src/main/scala/com.snowplowanalytics/iglu.schemaddl/jsonschema/json4s/StringSerializers.scala

```diff
@@ -23,15 +23,19 @@ object StringSerializers {
 
   object FormatSerializer extends CustomSerializer[Format](_ => (
     {
-      case JString(format) if format == "ipv4" => Ipv4Format
-      case JString(format) if format == "ipv6" => Ipv6Format
-      case JString(format) if format == "uri" => UriFormat
-      case JString(format) if format == "email" => EmailFormat
-      case JString(format) if format == "hostname" => HostNameFormat
-      case JString(format) if format == "date-time" => DateTimeFormat
-      case JString(format) if format == "date" => DateFormat
-      case JString(format) => CustomFormat(format)
-      case x => throw new MappingException("Format must be string")
+      case JObject(list) =>
+        list.find(_._1 == "format") match {
+          case Some(JsonAST.JField(_, JsonAST.JString(format))) if format == "ipv4" => Ipv4Format
+          case Some(JsonAST.JField(_, JsonAST.JString(format))) if format == "ipv4" => Ipv4Format
+          case Some(JsonAST.JField(_, JsonAST.JString(format))) if format == "ipv6" => Ipv6Format
+          case Some(JsonAST.JField(_, JsonAST.JString(format))) if format == "uri" => UriFormat
+          case Some(JsonAST.JField(_, JsonAST.JString(format))) if format == "email" => EmailFormat
+          case Some(JsonAST.JField(_, JsonAST.JString(format))) if format == "hostname" => HostNameFormat
+          case Some(JsonAST.JField(_, JsonAST.JString(format))) if format == "date-time" => DateTimeFormat
+          case Some(JsonAST.JField(_, JsonAST.JString(format))) if format == "date" => DateFormat
+          case Some(JsonAST.JField(_, JsonAST.JString(format))) => CustomFormat(format)
```

Comment:
Doesn't match `Some(JsonAST.JField(_, JsonAST.JInt(_)))` - I believe this is the reason `e3` spec doesn't work - it fails when sees `"maxLength": 32`.